src/desktop/space/mod.rs

```diff
@@ -613,18 +614,53 @@ impl Space {
 
                 // Then re-draw all windows & layers overlapping with a damage rect.
 
-                for element in layer_map
-                    .layers_on(WlrLayer::Background)
-                    .chain(layer_map.layers_on(WlrLayer::Bottom))
-                    .map(|l| l as &SpaceElem<R>)
+                for element in custom_elements
+                    .iter()
+                    .filter(|c| c.layer() == RenderLayer::Bottom)
+                    .map(|p| p as &SpaceElem<R>)
+                    .chain(
+                        layer_map
+                            .layers_on(WlrLayer::Background)
+                            .map(|l| l as &SpaceElem<R>)
+                            .chain(
+                                custom_elements
+                                    .iter()
+                                    .filter(|c| c.layer() == RenderLayer::AboveBackground)
+                                    .map(|c| c as &SpaceElem<R>),
+                            )
+                            .chain(layer_map.layers_on(WlrLayer::Bottom).map(|l| l as &SpaceElem<R>)),
+                    )
```

Comment:
```suggestion
                    )
                    .chain(
                        custom_elements
                            .iter()
                            .filter(|c| c.layer() == RenderLayer::AboveBackground)
                            .map(|c| c as &SpaceElem<R>),
                    )
                    .chain(layer_map.layers_on(WlrLayer::Bottom).map(|l| l as &SpaceElem<R>))
```

These two chains can be moved into the upper chain for consistency, no?